src/operator/contrib/tvmop/ufunc.cc

```diff
@@ -144,6 +144,118 @@ NNVM_REGISTER_OP(_backward_contrib_tvm_vadd)
     .set_attr<mxnet::FCompute>("FCompute<cpu>",
                                mxnet::op::TVMBinaryBackwardComputeUseNone<func_bakcward_vadd_cpu>);
 
+inline bool DegandradOpType(const nnvm::NodeAttrs& attrs,
+                            std::vector<int>* in_attrs,
+                            std::vector<int>* out_attrs) {
+  CHECK_EQ(in_attrs->size(), 1U);
+  CHECK_EQ(out_attrs->size(), 1U);
+
+  TYPE_ASSIGN_CHECK(*out_attrs, 0, in_attrs->at(0));
+  TYPE_ASSIGN_CHECK(*in_attrs, 0, out_attrs->at(0));
+  CHECK_LE(in_attrs->at(0), mshadow::kFloat64)
+    << "Only support float32 and float64.";
```

Comment:
You should have a different rule for checking this, the ids of data types could be easily changed and this rule will be broken.

Reply:
Changed to 
"CHECK(in_attrs->at(0) == mshadow::kFloat64 ||
               in_attrs->at(0) == mshadow::kFloat32)
    << "Only support float32 and float64.";"